tracee-rules/output.go

```diff
@@ -44,24 +47,53 @@ func setupOutput(resultWriter io.Writer, clock Clock, webhook string) (chan type
 			}
 
 			if webhook != "" {
-				payload, err := prepareJSONPayload(res)
-				if err != nil {
-					log.Printf("error preparing json payload for %v: %v", res, err)
-					continue
+				if err := sendToWebhook(res, webhook, webhookTemplate, realClock{}); err != nil {
+					log.Println(err)
 				}
-				resp, err := http.Post(webhook, "application/json", strings.NewReader(payload))
-				if err != nil {
-					log.Printf("error calling webhook for %v: %v", res, err)
-					continue
-				}
-				resp.Body.Close()
 			}
 		}
 	}()
 	return out, nil
 }
 
-func prepareJSONPayload(res types.Finding) (string, error) {
+func sendToWebhook(res types.Finding, webhook string, webhookTemplate string, clock Clock) error {
+	var payload string
+
+	switch {
+	case webhookTemplate != "":
+		t, err := template.New(filepath.Base(webhookTemplate)).
+			Funcs(map[string]interface{}{
+				"unixToRFC3339": func(unixTs float64) string {
+					return time.Unix(int64(unixTs), 0).UTC().Format("2006-01-02T15:04:05Z")
+				},
+			}).ParseFiles(webhookTemplate)
+		if err != nil {
+			return fmt.Errorf("error preparing webhook template: %v", err)
+		}
+
+		buf := bytes.Buffer{}
+		if err := t.Execute(&buf, res); err != nil {
+			return fmt.Errorf("error writing to the template: %v", err)
+		}
+		payload = buf.String()
+
+	default:
+		var err error
+		payload, err = prepareJSONPayload(res, clock)
+		if err != nil {
+			return fmt.Errorf("error preparing json payload: %v", err)
+		}
+	}
+
+	resp, err := http.Post(webhook, "application/json", strings.NewReader(payload))
```

Comment:
We can improve it further in a separate PR as you mentioned, but another way I can think of handling the current situation is making it `text/plain` type for now without complicating it too much. We don't send binary data anyway.

Reply:
mm but it is not plain text, is it? if you say that the content type doesn't matter (implied by the suggestion), why don't you just remove it? I think some servers will use this header to handle the payload differently, so IMO it's better for now to at least support json properly (which is the common use case), and if some server doesn't care about the content type and expects xml anyway, they wouldn't mind the header, and in the future we can let the use configure this (and other webhook settings)

Reply:
I went ahead and implemented content-type https://github.com/aquasecurity/tracee/pull/582/commits/df1b689633b48c92c7d7a3040dcb35dd16d35949 It will prevent any ambiguity.